File: 종복/1-1주차/타겟 넘버.js
// 1sol
// 5분

function solution(numbers, target) {
    var answer = 0;
    let n = numbers.length;
    
    function DFS(index, sum){
        
        if(index === n){
            if(sum === target){
                answer += 1;
            }
        }else{
           DFS(index+1, sum+numbers[index]);
           DFS(index+1, sum-numbers[index]);
        }
        
    }
    
    DFS(0, 0);
    
    return answer;
}

File: Java/프로그래머스/Level 2/DFS&BFS/타겟넘버.java
class Solution {
    
    static int answer = 0; 
    
    public void DFS(int sum, int pos, int[] numbers, int target){
        
        if(pos == numbers.length){
            if(sum == target){
                answer += 1;
            }
            return; 
        }else{
            DFS(sum+numbers[pos], pos+1, numbers, target);
            DFS(sum-numbers[pos], pos+1, numbers, target);
        }   
        
    }
    
    
    public int solution(int[] numbers, int target) {
       
        answer = 0; 
        
        DFS(0, 0, numbers, target);
     
        return answer;
    }
}
